enhancements/update/upgrade-without-registry.md

```diff
@@ -0,0 +1,989 @@
+      # operator will look for the upgrade bundle and will use it to determine the
+      # version, pin and pre-load the images, and finally then proceed to the
+      # upgrade as usual.
+      enabled: true|false
+
+      # Indicates if the cluster version operator should monitor devices
+      # to automatically detect when an upgrade bundle is available. The
+      # defaut is `false`.
+      monitor: true|false
+
+      # Indicates the location of the upgrade bundle when `monitor` is `false`.
+      file: /root/m-bundle.tar # Can be a file.
+      file: /dev/sdb # Or a device.
+
+      # This is an optional digest of the bundle. If set then the cluster version
+      # operator will ensure that the bundle is used only if it matches this
+      # digest. This is intended to prevent use of a wrong USB stick, or use of
+      # a perfectly fine USB stick in the wrong cluster. The administrator of
+      # the cluster sets it in advance, before the technician arrives in the
+      # cluster site and plugs the USB stick.
+      digest: sha256:....
+
+status:
+  desired:
+    bundle:
+      # Location of the upgrade bundle. This is populated by the monitor, if
+      # enabled, or copied from the `spec.desiredUpdate.bundle.file` field if
+      # monitoring is disabled.
+      file: /root/my-bunde.tar # Can be a file.
+      file: /dev/sdb # Or a device.
+
+      # Node where the bundle is available. This is optional. By default the
+      # cluster version operator will look for the bundle in all the nodes, but
+      # if this is used it will look only in the given node.
+      node: node2
+
+      # Contains the current status of the bundle processing in each node
+      # of the cluster. Keys are the node names.
+      nodes:
+        node0:
+          # Indicates if the upgrade bundle has already been extracted to
+          # the `/var/lib/upgrade/4.13.7-x86_64` directory.
+          extracted: true|false
+          # Indicates if the images from the upgrade bundle have been already
+          # copied to the `/etc/containers/storage` directory.
+          loaded: true|false
+          # Contains the metadata of the bundle as seen by this node. This is
+          # taken from the `metadata.json` file that is part of the bundle.
+          metadata: '{ ... }'
+        node1: { ... }
+        ...
+```
+
+And this is a summary of the new `ContainerRuntimeConfig` fields:
+
+```yaml
+apiVersion: machineconfiguration.openshift.io/v1
+kind: ContainerRuntimeConfig
+metadata:
+  name: pin-upgrade
+spec:
+  containerRuntimeConfig:
+
+    # Additional image store directories. Translates into something like this
+    # inside the `/etc/containers/storage.conf` file:
+    #
+    # additionalimagestores = [
+    #   "/var/lib/my-images/...",
+    #   "/var/lib/your-images/...",
+    #   ...
+    # ]
+    additionalImageStores:
+    - /var/lib/my-images
+    - /var/lib/my-images
```

Comment:
Also, I don't understand why we have an `additionalImageStores` at all here, isn't all we need covered by the image pinning?

Reply:
The example is incorrect, should have been two different directories, I will fix that.

It is a list in order to support multiple directories like the `additionalimagestores` setting in `storage.conf` does.

With the current approach (based on using the equivalent of `crictl pull` to copy the images) this isn't necessary. It would be necessary if we change the approach and decide to just copy the files (in `containers-storage` format). I am keeping this here because it may be useful for other use cases. I can remove it if you think it is too confusing.